src/components/PostDetails.tsx

```diff
@@ -1,106 +1,123 @@
-import React from 'react';
+import React, { useContext, useEffect } from 'react';
+import { UserListContext } from './listContext';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { deleteComment } from './todosApi/api-todos';
 
 export const PostDetails: React.FC = () => {
+  const {
+    errorComments,
+    comments,
+    post,
+    selectedPostId,
+    loaderDetails,
+    buttonAddComment,
+    setButtonAddComment,
+    addComment,
+    setAddComent,
+    setButtonLoading,
+    fetchUserComments,
+    setComments,
+  } = useContext(UserListContext);
+
+  const selectedPost = post.find(po => po.id === selectedPostId);
+
+  const handleAddComment = () => {
+    setAddComent(prev => !prev);
+    setButtonAddComment(false);
+    setButtonLoading(false);
+  };
+
+  useEffect(() => {
+    if (selectedPostId) {
+      fetchUserComments(selectedPostId);
+    }
+  }, [selectedPostId]);
```

Comment:
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/86923089/3d45af0f-2b74-4a05-9744-04b017e4a55b)

just add it to the dependencies array, but wrap that function into useCallback in ListContext